subcommand/connect-init/command.go

```diff
@@ -81,19 +99,69 @@ func (c *Command) Run(args []string) int {
 			return 1
 		}
 	}
+	// First do the ACL Login, if necessary.
+	if c.flagACLAuthMethod != "" {
+		// Validate flags related to ACL login.
+		if c.flagMeta == nil {
+			c.UI.Error("-meta must be set")
+			return 1
+		}
```

Comment:
No worries!

Meta is required to `ConsulLogin()` but not `consulClient.Agent().ACL().Login()`, in order to preserve our original code paths it is required,  as we have always passed `-meta pod=default/podName` .(https://github.com/hashicorp/consul-k8s/blob/c70361e43ce8ad6801c0fef7b2119d9dca891ac0/connect-inject/container_init.go#L425)

Reply:
now that we require `-pod-namespace` and `-pod-name` and that's all that `-meta` is used for, can we get away without it now and instead use those flags?

Reply:
I suppose, I'd have to back out all the extra flag code I added to support meta taking in a map after we decided that was mandatory in the last PR.

Right now because of *where* the command is run there isn't support for adding any extra `-meta` fields but I wonder if that is something we might want in the future? I'm not sure if that is desirable since it's just ACL Login.

One alternative could be to not use it now, hardcode the ACL Login meta field, and repurpose this field as extra meta to be sent to envoy bootstrapping once we get an API for it?

Reply:
OK, I think I see a good path forward here.
1. I dont think that `-meta` is a meaningful flag for ACL Login, so we shouldn't use it for that.
2. I dont see a near-term use for `-meta` for the rest of the command, or for envoy bootstrapping, etc

So let's keep the new flags map code and remove the meta flag, using the `-pod-name/space` flags to generate our old `-meta` options for ACL Login?

Reply:
that sounds good to me!

Reply:
sounds good!